### DISCUSSION TITLE
Catching raw Ethernet frames on pico W

### DISCUSSION THREAD

**Discussion by sapfeer0k** [micropython/micropython#13279] [RP2040 / Pico]
Hi everyone,

I am trying to implement a Wake-On-Lan catcher using pico w. Most of the software uses UDP port 7/9 for wake up and I can easily receive them using udp socket .

The problem starts with apple AirPlay , their devices send raw Ethernet frame , that can't be seen above second OSI level. Here is an example from `tcpdump` (I replaced my target device mac address with zeros):
```
17:29:38.201520 31:a1:45:25:ef:18 > ff:ff:ff:ff:ff:ff, ethertype Unknown (0x0842), length 122:
	0x0000:  ffff ffff ffff 0000 0000 0000 0000 0000  ................
	0x0010:  0000 0000 0000 0000 0000 0000 0000 0000  ................
	0x0020:  0000 0000 0000 0000 0000 0000 0000 0000  ................
	0x0030:  0000 0000 0000 0000 0000 0000 0000 0000  ................
	0x0040:  0000 0000 0000 0000 0000 0000 0000 0000  ................
	0x0050:  0000 0000 0000 0000 0000 0000 0000 0000  ................
	0x0060:  0000 0000 0000 0000 0000 0000 
```
I found lwip extension in the micropython repo, but I have exceptions during the setup of a raw socket . And there is no example available (at least I couldn't find any) . If someone can suggest or show example how could I catch those raw packets produced by apple software , it would be very helpful.

**Comment by sapfeer0k** [comment]
I've managed to find ICMP example python script, adapted it and now I am even able to receive ARP frames sent by arping, but still I can't see any packages addressed to broadcast (`ff:ff:ff:ff:ff:ff`) . Seems to be they are getting filtered on the driver level :(

**Reply by jimmo** [reply]
@sapfeer0k  I don't think this is supported, but I'm unsure off the top of my head exactly which level it gets filtered out.

The easiest first step would be to enable tracing on the cyw43 driver and see if the packet arrives.

```py
wlan = network.WLAN(network.STA_IF)
wlan.config(trace=x)
```

where `x` is a bitfield of trace flags:

```py
CYW43_TRACE_ASYNC_EV = const(0x0001)
CYW43_TRACE_ETH_TX = const(0x0002)
CYW43_TRACE_ETH_RX = const(0x0004)
CYW43_TRACE_ETH_FULL = const(0x0008)
CYW43_TRACE_MAC = const(0x0010)
```

You probably want `CYW43_TRACE_ETH_RX | CYW43_TRACE_ETH_FULL`
